src/components/HomePage.jsx

```diff
@@ -4,20 +4,80 @@ import { useNavigate } from "react-router-dom";
 import ApiRequests from "./ApiRequests";
 import Categories from "./Categories";
 const HomePage = () => {
-  const { user } = useContext(UserContext);
+  const { user, setUser } = useContext(UserContext);
+  const [inputValue, setInputValue] = useState("");
+  const [isLoading, setIsLoading] = useState(true);
+  const [allUsers, setAllUsers] = useState("");
+  const [information, setInformation] = useState("");
   const navigate = useNavigate();
-
+  useEffect(() => {
+    ApiRequests.getUsers().then((res) => {
+      setAllUsers(res.data.users);
+      setIsLoading(false);
+    });
+  }, []);
+  const logIn = () => {
+    let found = false;
+    let count = 0;
+    allUsers.map((data, index) => {
+      if (data.username === inputValue) {
+        console.log(count);
+        found = true;
+        return setUser(data);
+      } else if (allUsers.length - 1 === index && !found) {
+        console.log("not found");
+        found = true;
+        return setInformation("user not found");
+      }
+    });
+    setInputValue("");
+  };
+  const logOut = () => {
+    setUser(null);
+  };
+  if (isLoading) return <>loading...</>;
   return (
     <div>
-      <h1>LARGEST GAMES REVIEW SITE!</h1>
+      <h1>LARGEST GAMES REVIEW WEBSITE!</h1>
       <button
         onClick={() => {
-           navigate("/reviews");
+          navigate("/reviews");
         }}
       >
         View all reviews
       </button>
       <Categories />
+      {!user ? (
+        <>
+          <input
+            type="text"
+            value={inputValue}
+            onChange={(e) => {
+              setInformation("");
+              setInputValue(e.target.value);
+            }}
+          />
+          <button onClick={logIn}>log in</button>
+          <div>user names allowed:</div>
+          <div>
+            {allUsers.map((user, index) => {
```

Comment:
I'd probably put your allowed usernames in your readme for people who are trying out your site. The functionality is good but there's no need for them all to be listed on the page.